README.md

```diff
@@ -227,13 +229,39 @@ val fetch: List[Int] => Future[List[User]] = usersService.fetch _
 val usersSource = Clump.source(fetch)(_.id)
 ```
 
-The ```Clump.sourceFrom``` method accepts a function that return a ```Map``` with the values for the found inputs.
+It is possible to create sources that have additional inputs, but the compiler isn't capable of inferring the input type for these cases. The solution is to use an explicit generic:
+
+```scala
+val fetch: (UserSession, List[Int]) => Future[List[User]] = usersService.fetch _
+def usersSource(session: UserSession) = 
```

Comment:
I think it is important to have the types explicitly but changed to not use an anonymous function. How do you like it now?